File: TrabajoI.C/C#/interfaz/ClaseD.cs
public class ClaseD : IOperable
{
    public int Numero { get; set; }

    public D(int numero)  // Constructor normal
    {
        Numero = numero;
        Console.WriteLine($"Constructor normal en D: {numero}");
    }

    public D(D original)  // Constructor copia
    {
        Numero = original.Numero;
        Console.WriteLine($"Constructor copia en D: {Numero}");
    }

    public void Operar()
    {
        Console.WriteLine("Operando en D");
    }
}

File: TrabajoI.C/Java/interfaz/ClaseD.java
package interfaz;

public class ClaseD implements Operable {
    private int num;

    public ClaseD(int num) {
        this.num = num;
    }

    // Constructor copia
    public ClaseD(ClaseD original) {
        this.num = original.num;
        System.out.println("Constructor copia en D: " + num);
    }

    @Override
    public void operar() {
        System.out.println("Operando en D");
    }   
}
